xds/src/main/java/io/grpc/xds/AddressFiltering.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright 2020 The gRPC Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.grpc.xds;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import io.grpc.Attributes;
+import io.grpc.EquivalentAddressGroup;
+import io.grpc.NameResolver.ResolutionResultAttr;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import javax.annotation.Nullable;
+
+final class AddressFiltering {
+  @ResolutionResultAttr
+  private static final Attributes.Key<PathChain> PATH_CHAIN_KEY =
+      Attributes.Key.create("io.grpc.xds.AddressFilter.PATH_CHAIN_KEY");
+
+  // Prevent instantiation.
+  private AddressFiltering() {}
+
+  /**
+   * Sets a path filter to an EquivalentAddressGroup. Any preexisting path filter will be discarded.
+   */
+  static EquivalentAddressGroup setPathFilter(EquivalentAddressGroup address, List<String> names) {
+    checkNotNull(address, "address");
```

Comment:
Those two checkers are not useful, it's redundant code.  If the arguments are `null`, it breaks immediately, same as what the checkers are doing. Same for below.

Reply:
The purpose of the checker is to provide a message (no matter it's a dummy string) rather than only a stacktrace. Because from the bug reporter's stacktrace you can only guess which field is null from the line number in the code, but the line number might be changing from commit to commit.  NPE in the stacktrace sometime does not infer which object is null. For example, in
```java
address.getAttributes().get(foo).equals()
```
The null object could be (theoretically) `address`, `address.getAttributes()` or `address.getAttributes().get(foo)`

Reply:
Whatever, not a big issue, I don't mind if you leave it there. Although, IMO, this kind of code is redundant. BTW, you do get different stacktrace for each part of `address.getAttributes().get(foo).equals()` being `null` and you can mostly tell what's wrong from there.

Reply:
>you do get different stacktrace 

Not before Java 14 https://openjdk.java.net/jeps/358

Reply:
Whatever, I'm not going to make simple things complex.